testutil/context.go

```diff
@@ -0,0 +1,44 @@
+package testutil
+
+import (
+	"context"
+	"log/slog"
+	"testing"
+
+	"github.com/authenticvision/util-go/logutil"
+)
+
+type contextConfig struct {
+	logLevel slog.Level
+}
+
+type ContextOption func(config *contextConfig)
+
+func WithLogLevel(level slog.Level) ContextOption {
+	return func(config *contextConfig) {
+		config.logLevel = level
+	}
+}
+
+// Context provides a logger for t.Context(). It might eventually provide
+// additional context values offered by util-go, so all code that uses util-go
+// context-specific code should get a context from this method in tests.
+func Context(t *testing.T, opts ...ContextOption) context.Context {
```

Comment:
`Context` is too undescriptive. `SetupLogger` or `ConfigureLogger` would be better

Reply:
I see your point, how about `SetupLogContext` because it it configures a context primarily, not a logger?

The idea behind naming it Context was to have a generic abstraction here that just gives us a test context, though that's probably another case of yagni, being more descriptive and renaming it later if need be works for me too.

Reply:
I was thinking more it should do more generic test setup we use, like also setting the default logger.
Just `testutil.Setup` might do then, because the context part is obvious from the return value

Reply:
After offline bikeshedding, we didn't come up with anything better than `Context` 🤷